### ISSUE TITLE
Method incorrect declaration

### ISSUE THREAD

**Issue by ajcvickers** [aspnet/Identity#1382]
Moving from https://github.com/aspnet/EntityFrameworkCore/issues/9500 by @roonikz
When you want to use IdentityBuilder to setup Identity services for your application (e.g. AddUserStore<>() AddRoleStore<>() AddPasswordValidator<>() and so on). When you have looked at these methods' declarations, you will see: 

```c#
    public virtual IdentityBuilder AddUserStore<TUser>() where TUser : class
    {
      return this.AddScoped(typeof (IUserStore<>).MakeGenericType(this.UserType), typeof (TUser));
    }
```

Looking at _**TUser**_ type, you are thinking about passing User type as generic parameter. Using Autofac DI container integration I have got this ArgumentException:

```
ArgumentException: The type 'CMS.Domain.User' is not assignable to service 'Microsoft.AspNetCore.Identity.IUserStore`1[[CMS.Domain.User, CMS.Domain, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'.
```
Then I decompile AddScoped() method, and here what I see:

```c#
    private IdentityBuilder AddScoped(Type serviceType, Type concreteType)
    {
      this.Services.AddScoped(serviceType, concreteType);
      return this;
    }
```
AddScoped method second argument is concreteType! As I think, it would be better to replace **`Add*WhateverIdentityService*<TUser> where TUser: class`**  by **`Add*WhateverIdentityService*<TWhateverIdentityService> where TWhateverIdentityService: class`** to avoid disagreements. Thank you.

### Further technical details
EF Core version: EF Core 2.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 
IDE: Visual Studio 2017

**Comment by rooonikz** [comment]
@HaoK I think it still has some stuff to be done. As i mentioned before, this renamings should be applied to all Identity services. As i see, AddPasswordValidator, AddRoleValidator and AddUserValidator generic parameters declarations are invaid. Thx!

**Comment by HaoK** [comment]
Ah right, I'll rename the others today

**Comment by HaoK** [comment]
Fixed the rest bc75b1a0e73d782fbe221fb6f12855d1a337a5b1
